### PR TITLE
support in subquery and compare subquery.

### PR DESCRIPTION
**Pull request by hanfei19910905**
@shenli @coocood @zimulala  PTAL

### CODE REVIEW

File: executor/new_executor.go [hunk 1]
```diff
@@ -851,6 +851,37 @@ type ApplyExec struct {
 	Src         Executor
 	outerSchema expression.Schema
 	innerExec   Executor
+	checker     *conditionChecker
+}
+
+type conditionChecker struct {
```
Comment (shenli) [inline executor/new_executor.go]: Add comment for this.

File: executor/new_executor.go [hunk 2]
```diff
@@ -851,6 +851,37 @@ type ApplyExec struct {
...
+		if err != nil {
+			return nil, false, errors.Trace(err)
+		}
+		if matched {
+			c.matched = true
+		}
+	} else if c.all {
+		matched = c.matched
+	}
+	if (matched && !c.all) || (!matched && c.all) || lastRow {
```
Comment (shenli) [inline executor/new_executor.go]: matched != c.all is enough?

File: executor/new_executor.go [hunk 3]
```diff
@@ -877,17 +911,31 @@ func (e *ApplyExec) Next() (*Row, error) {
...
-	outerRow, err := e.innerExec.Next()
-	e.innerExec.Close()
-	if err != nil {
-		return nil, errors.Trace(err)
+	for {
+		for _, col := range e.outerSchema {
+			idx := col.Index
+			col.SetValue(&srcRow.Data[idx])
+		}
+		outerRow, err := e.innerExec.Next()
```
Comment (shenli) [inline executor/new_executor.go]: innerRow

**Comment by shenli** [comment]
LGTM

File: executor/new_executor.go [hunk 4]
```diff
@@ -877,17 +912,31 @@ func (e *ApplyExec) Next() (*Row, error) {
...
+			return nil, errors.Trace(err)
+		}
+		if innerRow != nil {
+			srcRow.Data = append(srcRow.Data, innerRow.Data...)
+		}
+		if e.checker == nil {
+			e.innerExec.Close()
+			return srcRow, nil
+		}
+		resultRow, finished, err := e.checker.Exec(srcRow, innerRow == nil)
```
Comment (coocood) [inline executor/new_executor.go]: Handle the case 'innerRow == nil' outside of checker.Exec would be cleaner.

**Comment by coocood** [comment]
LGTM

**Pull request merged**